mentoring/google-summer-of-code.md

```diff
@@ -0,0 +1,78 @@
+# Google Summer of Code
+
+Kubernetes is happy to announce that we have applied to participate in the Google Summer of Code (GSoC) program
+under the Cloud Native Computing Foundation (CNCF), running from January 2018 to August 2018.
+
+Please see the [main program page](https://summerofcode.withgoogle.com/) for general information about the program,
+such as its purpose, timeline, eligibility requirements, and how to apply.
+
+## Schedule
+
+The following schedule is taken from the official [timeline](https://summerofcode.withgoogle.com/how-it-works/#timeline).
+
+* January 4: Organization applications open
+* January 23: Organization applications deadline
+* February 12: Google will publish the accepted mentoring organizations
+* March 12: Student applications open
+* March 27: Student applications deadline
+* March 27 - April 23: Applications are reviewed
+* April 23: Accepted students are announced
+* April 23 - May 14: Community Bonding period
+* May 14 - August 6: Coding period
+* June 11 - June 15: First evaluations
+* July 9 - July 13: Second evaluations
+* August 6 - August 14: Students submit code and final evaluations
+* August 14 - August 21: Mentors submit final evaluations
+* August 22: Results announced
+
+## What is Kubernetes?
+
+Kubernetes is an open-source system for automating deployment, scaling, and management of containerized applications.
+It groups containers that make up an application into logical units for easy management and discovery. Kubernetes builds upon 15 years of experience of running production workloads at Google, combined with best-of-breed ideas and practices from the community. Please see [kubernetes.io](https://kubernetes.io/) for more details.
+
+## What are SIGs?
+
+Kubernetes is a set of projects, each shepherded by a special interest group (SIG). To get a grasp of the projects that we work on, check out the complete [list of SIGs](/sig-list.md).
+
+This [slide deck](https://docs.google.com/presentation/d/1JqcALpsg07eH665ZXQrIvOcin6SzzsIUjMRRVivrZMg) provides an overview of how
+the Kubernetes community is organized.
+
+## Communication
+
+- Mailing List: development announcements and discussions appear on the Google group `kubernetes-dev` (send mail to kubernetes-dev@googlegroups.com).
+- Slack: Please [sign up](http://slack.k8s.io/) for an invite.
+
+## Contribute
+
+As part of the application process, the Google Summer of Code program recommends that candidates make small contributions to the project they intend to apply for.
+To start working on the project, make sure to fill out the [CLA](../CLA.md  ) and check if you have the right environment with this guide. The README in the [community repo](https://github.com/kubernetes/community) details these things and more.
```

Comment:
A CLA link such as [CLA](/CLA.md) might make the file itself more movable.

Reply:
@stp-ip updated. ptal, thanks.